core/src/main/scala/org/apache/spark/storage/ShuffleBlockFetcherIterator.scala

```diff
@@ -272,7 +275,7 @@ final class ShuffleBlockFetcherIterator(
 
   override def hasNext: Boolean = numBlocksProcessed < numBlocksToFetch
 
-  override def next(): (BlockId, Try[Iterator[Any]]) = {
+  override def next(): (BlockId, Try[InputStream]) = {
```

Comment:
I just started writing a big comment about why I thought this was unnecessary because I thought the `TaskCompletionListener` would take care of it for us ... and then I realized that this was necessary so that we release buffers as soon as finish with them as we fetch a bunch of blocks.

would you mind adding a small doc here that callers of this class should always be sure to wrap the results in a `CompletionIterator` which closes the `InputStream`, to be sure buffers are released as soon as possible?  Would help callers and future readers of this code